src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -191,31 +191,59 @@ public CaseEvent createCaseEvent(final CaseEvent caseEvent, final Case newCase,
       // should we create an ad hoc action?
       createAdHocAction(category, caseEvent);
 
-      // transition case group status
-      CaseGroup caseGroup = caseGroupRepo.findOne(targetCase.getCaseGroupFK());
-      try {
-        caseGroupService.transitionCaseGroupStatus(caseGroup, caseEvent.getCategory(), targetCase.getPartyId());
-      } catch (CTPException e) {
-        //The transition manager throws an exception if the event doesn't cause a transition, however there are lots of
-        // events which do not cause CaseGroupStatus transitions, (this is valid behaviour).
-        log.debug(e.getMessage());
-      }
+      transitionCaseGroupStatus(targetCase, caseEvent);
 
       // if this is a respondent enrolling event
       if (caseEvent.getCategory().toString().equals("RESPONDENT_ENROLED")) {
         // are there other case groups that need updating
         List<CaseGroup> caseGroups = caseGroupService.transitionOtherCaseGroups(targetCase);
         checkCaseState(category, caseGroups, caseEvent, newCase);
+
       } else {
+
         // should a new case be created?
         createNewCase(category, caseEvent, targetCase, newCase);
-        effectTargetCaseStateTransition(category, targetCase);
+
+        if (caseEvent.getCategory().toString().equals("SUCCESSFUL_RESPONSE_UPLOAD")) {
```

Comment:
Just a quick question - would this need to handle `COMPLETED_BY_PHONE` category as well?

Reply:
Didn't think of that.